docs/operate/reference/advanced-modules/_index.md

```diff
@@ -46,6 +46,74 @@ Running {{< glossary_tooltip term_id="modular-resource" text="modular resources"
 
 However, if you are unable to use modular resources because you need to host `viam-server` on a non-Linux system or have an issue with compilation, you may need to [implement a custom component and register it on a server configured as a remote](/operate/reference/advanced-modules/custom-components-remotes/) on your machine.
 
+## Package and deploy using Docker
+
+In rare cases, you may need to package and deploy a module using Docker.
+Use cases for this include:
+
+- Your module has complex system dependencies that cannot be installed on a machine.
+- You have a large bundle and want to use layer caching to reduce the size of the download.
+- You have specific security requirements that are difficult to meet with the default module deployment.
+
+If you choose to deploy your module using Docker, we recommend creating a first run script or binary to run any necessary setup steps.
+
+{{% expand "Click for first run script instructions" %}}
+
+1. Create a tarball that contains:
+
+   - The module's entrypoint script or binary
+   - A <file>meta.json</file>
+   - A first run script or binary that will be executed during the setup phase, for example:
+
+     ```sh {id="terminal-prompt" class="command-line" data-prompt="$"}
+     #!/usr/bin/env bash
+
+     if [[ -n "$VIAM_TEST_FAIL_RUN_FIRST" ]]; then
```

Comment:
I'm not familiar with this env var. Is it supposed to be in all first_run scripts? What does it do? Maybe I should add it to mine... :flushed:

Reply:
Great question--just pulled this in from the setup phase guide on confluence which in turn links to https://github.com/viam-labs/wifi-sensor/blob/7823b6ad3edcbbbf20b06c34b3181453f5f3f078/first_run.sh

Reply:
I also am unclear on what this is for here. Can you add an explanation in? Does this get set to true if a previous run failed? Or is this jsut for testing?

Reply:
In the RDK, this env var is mentioned in 2 places: a test and an example first_run.sh. I have cloned about 60 Viam-related repos locally, and it's not mentioned in any of the others. I've searched Github for it, and can only find copies of those 2 locations. I wonder if there was some plan to add this in, but it never happened and these are just the vestiges of a feature that never was.

I think we should remove this paragraph from the example.

Reply:
Thanks for looking around. Will remove.